Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -305,6 +284,36 @@ extension NIOSSLCertificate {
 
         return certificates
     }
+
+    /// Calls the given body function with a temporary buffer containing the DER-encoded bytes of this
+    /// certificate. This function does allocate for these bytes, but there is no way to avoid doing so with the
+    /// X509 API in BoringSSL.
+    ///
+    /// The pointer provided to the closure is not valid beyond the lifetime of this method call.
+    private func withUnsafeDERCertificateBuffer<T>(_ body: (UnsafeRawBufferPointer) throws -> T) throws -> T {
```

Comment:
this should be `rethrows` I think and we should remove the `throw NIOSSLError.unableToAlloc` I think